sequencer/src/state.rs

```diff
@@ -402,83 +406,110 @@ impl<'a> StateCatchup for SqlStateCatchup<'a> {
+        .expect_ok()
+        .context("Index not found in block merkle tree")?;
+    let path = <u64 as ToTraversalPath<3>>::to_traversal_path(
+        &(block_number - 1),
+        block_merkle_tree.height(),
+    );
 
-        let Delta { fees_delta } = delta;
+    storage
+        .store_state::<BlockMerkleTree, 3>(proof.proof, path, block_number)
+        .await
+        .context("failed to insert merkle nodes for block merkle tree")?;
 
-        // Insert fee merkle tree nodes
-        for delta in fees_delta {
-            let (_, proof) = fee_merkle_tree
-                .universal_lookup(delta)
-                .expect_ok()
-                .context("Index not found in fee merkle tree")?;
-            let path: Vec<usize> = <FeeAccount as ToTraversalPath<256>>::to_traversal_path(
-                &delta,
-                fee_merkle_tree.height(),
-            );
+    storage
+        .set_last_state_height(block_number as usize)
+        .await
+        .context("failed to set last state height")?;
+    storage.commit().await?;
 
-            storage
-                .write()
-                .await
-                .store_state::<FeeMerkleTree, 256>(proof.proof, path, block_number)
-                .await
-                .context("failed to insert merkle nodes for block merkle tree")?;
-        }
+    Ok(())
+}
 
-        if block_number == 0 {
-            continue;
-        }
+pub async fn update_state_storage_loop(
+    mut storage: SqlDataSource<SeqTypes, Provider>,
+    instance: NodeState,
+) -> anyhow::Result<()> {
+    // get last saved merklized state
 
-        // Insert block merkle tree nodes
-        let (_, proof) = block_merkle_tree
-            .lookup(block_number - 1)
-            .expect_ok()
-            .context("Index not found in block merkle tree")?;
-        let path = <u64 as ToTraversalPath<3>>::to_traversal_path(
-            &(block_number - 1),
-            block_merkle_tree.height(),
-        );
+    let last_height = storage.get_last_state_height().await?;
 
-        storage
-            .write()
-            .await
-            .store_state::<BlockMerkleTree, 3>(proof.proof, path, block_number)
-            .await
-            .context("failed to insert merkle nodes for block merkle tree")?;
+    let mut parent_leaf = storage.get_leaf(last_height).await.resolve().await;
+    let mut leaves = storage.subscribe_leaves(last_height + 1).await;
 
-        storage
-            .write()
+    while let Some(leaf) = leaves.next().await {
+        loop {
+            if update_state_storage(
+                &mut storage,
+                instance.clone(),
+                parent_leaf.clone(),
+                leaf.clone(),
+            )
             .await
-            .set_last_state_height(block_number as usize)
-            .await?;
+            .is_err()
+            {
+                tracing::error!(
+                    "failed to update state storage for leaf {:?}. retrying..",
```

Comment:
We should include the error in this log. Also probably a small delay before retrying would be good